packages/core/src/amazonq/webview/messages/messageDispatcher.ts

```diff
@@ -60,12 +60,19 @@ export function dispatchWebViewMessagesToApps(
                         source: msg.trigger,
                         result: 'Succeeded',
                     })
+                    return
                 } else if (isClickTelemetry(msg)) {
                     telemetry.ui_click.emit({
                         elementId: msg.source,
                         result: 'Succeeded',
                     })
+                    return
                 }
+                return
+            }
+            case 'disclaimer-acknowledged': {
+                void globals.globalState.tryUpdate('aws.amazonq.disclaimerAcknowledged', true)
```

Comment:
tryUpdate is synchronous
```suggestion
               globals.globalState.tryUpdate('aws.amazonq.disclaimerAcknowledged', true)
```